app/models/user_card.rb

```diff
@@ -85,13 +85,10 @@ def set_user_card_name
   #
   # @see MonthYear#next_month_this
   #
-  # FIXME: in seeds, I found a card_transaction that had a due date a year later than it should
-  # see #set_month_year, it only sets the month and year on create
   def set_current_dates
     return false if current_due_date.nil? || days_until_due_date.nil?
 
-    today = Date.current
-    self.current_due_date = next_month_this(day: current_due_date.day) if today >= current_due_date
+    self.current_due_date = next_month_this(day: current_due_date.day)
```

Comment:
Regardless of the current_due_date being in the past or in the future, it has to be in the next_month.